code/modules/mob/living/simple_animal/borer/borer.dm

```diff
@@ -0,0 +1,1023 @@
+		return
+
+	src << "<span class='warning'>You slither up [H] and begin probing at their ear canal...</span>"
+	src.layer = MOB_LAYER
+	if(!do_mob(src, H, 30))
+		src << "<span class='warning'>As [H] moves away, you are dislodged and fall to the ground.</span>"
+		return
+
+	if(!H || !src)
+		return
+
+	Infect(H)
+
+/mob/living/simple_animal/borer/proc/Infect(mob/living/carbon/C)
+	if(!C)
+		return
+
+	if(C.has_brain_worms())
+		src << "<span class='warning'>[C] is already infested!</span>"
+		return
+
+	if(!C.key || !C.mind)
+		src << "<span class='warning'>[C]'s mind seems unresponsive. Try someone else!</span>"
+		return
+
+	if (C && C.dna && istype(C.dna.species, /datum/species/skeleton))
+		src << "<span class='warning'>[C] does not possess the vital systems needed to support us.</span>"
+		return
+
+	src.victim = C
+	forceMove(victim)
+
+	RemoveBorerActions()
+	GrantInfestActions()
+
+	log_game("[src]/([src.ckey]) has infested [victim]/([victim.ckey]")
+
+/mob/living/simple_animal/borer/verb/secrete_chemicals()
+	set category = "Borer"
+	set name = "Secrete Chemicals"
+	set desc = "Push some chemicals into your host's bloodstream."
+
+	if(!victim)
+		src << "<span class='warning'>You are not inside a host body.</span>"
+		return
+
+	if(stat != CONSCIOUS)
+		src << "<span class='warning'>You cannot secrete chemicals in your current state.</span>"
+
+	if(docile)
+		src << "<span class='warning'>You are feeling far too docile to do that.</span>"
+		return
+
+	var content = ""
+	content += "<p>Chemicals: <span id='chemicals'>[chemicals]</span></p>"
+
+	content += "<table>"
+
+	for(var/datum in typesof(/datum/borer_chem))
+		var/datum/borer_chem/C = new datum()
+		if(C.chemname)
+			content += "<tr><td><a class='chem-select' href='?_src_=\ref[src];src=\ref[src];borer_use_chem=[C.chemname]'>[C.chemname] ([C.chemuse])</a><p>[C.chem_desc]</p></td></tr>"
+
+	content += "</table>"
+
+	var/html = get_html_template(content)
+
+	usr << browse(null, "window=ViewBorer\ref[src]Chems;size=600x800")
+	usr << browse(html, "window=ViewBorer\ref[src]Chems;size=600x800")
+
+	return
+
+/mob/living/simple_animal/borer/verb/hide()
+	set category = "Borer"
+	set name = "Hide"
+	set desc = "Become invisible to the common eye."
+
+	if(victim)
+		src << "<span class='warning'>You cannot do this while you're inside a host.</span>"
+
+	if(stat != CONSCIOUS)
+		return
+
+	if (!hiding)
+		layer = LATTICE_LAYER
+		visible_message("<span class='name'>[src] scurries to the ground!</span>", \
+						"<span class='noticealien'>You are now hiding.</span>")
+		hiding = TRUE
+	else
+		layer = MOB_LAYER
+		visible_message("[src] slowly peaks up from the ground...", \
+					"<span class='noticealien'>You stop hiding.</span>")
+		hiding = FALSE
+
+/mob/living/simple_animal/borer/verb/dominate_victim()
+	set category = "Borer"
+	set name = "Paralyze Victim"
+	set desc = "Freeze the limbs of a potential host with supernatural fear."
+
+	if(world.time - used_dominate < 150)
```

Comment:
Why the two cooldown checks differ ? Also one should be checked after the input, otherwise you could stack them.